flink-ml/flink-ml-api/src/main/java/org/apache/flink/ml/api/misc/param/Params.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.api.misc.param;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The map-like container class for parameter. This class is provided to unify the interaction with
+ * parameters.
+ */
+@PublicEvolving
+public class Params implements Serializable {
+	private final Map<String, Object> paramMap = new HashMap<>();
+
+	/**
+	 * Returns the value of the specific parameter, or default value defined in the {@code info} if
+	 * this Params doesn't contain the param.
+	 *
+	 * @param info the info of the specific parameter, usually with default value
+	 * @param <V>  the type of the specific parameter
+	 * @return the value of the specific parameter, or default value defined in the {@code info} if
+	 * this Params doesn't contain the parameter
+	 * @throws RuntimeException if the Params doesn't contains the specific parameter, while the
+	 *                          param is not optional but has no default value in the {@code info}
+	 */
+	@SuppressWarnings("unchecked")
+	public <V> V get(ParamInfo<V> info) {
+		V value = (V) paramMap.getOrDefault(info.getName(), info.getDefaultValue());
+		if (value == null && !info.isOptional() && info.getDefaultValue() == null) {
```

Comment:
This is a bit confusing from your previous `hasDefaultValue()` javadoc. if `hasDefaultValue()` is added and here defaultValue cannot be null if it is not optional. it seems like:
```
info.hasDefaultValue() === (!info.isOptional() && info.getDefaultValue() == null)
```

Reply:
In my opinion, null may be a valid default value. Some estimator may treat null value as a special case. It works like this: when getting a param which is not configured,
if hasDefaultValue, use defaultValue
if !hasDefaultValue && isOptional, use null
if !hasDefaultValue && !isOptional, throw exception, this means the params must be set manually
I will add @Nullable at the setter of defaultValue. Or do you have any question about this behavior or this prerequisite(null can be valid param value)?